src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -3643,35 +3652,31 @@ bool Creature::IsNotReachableAndNeedRegen() const
     return false;
 }
 
-time_t Creature::GetLastDamagedTime() const
+std::shared_ptr<time_t> const& Creature::GetLastLeashExtensionTimePtr() const
 {
-    if (!_lastDamagedTime)
-        return time_t(0);
+    if (m_lastLeashExtensionTime == nullptr)
+        m_lastLeashExtensionTime = std::make_shared<time_t>(time(nullptr));
+    return m_lastLeashExtensionTime;
+}
 
-    return *_lastDamagedTime;
+void Creature::SetLastLeashExtensionTimePtr(std::shared_ptr<time_t> const& timer)
+{
+    m_lastLeashExtensionTime = timer;
 }
 
-std::shared_ptr<time_t> const& Creature::GetLastDamagedTimePtr() const
+void Creature::ClearLastLeashExtensionTimePtr()
 {
-    return _lastDamagedTime;
+    m_lastLeashExtensionTime.reset();
 }
 
-void Creature::SetLastDamagedTime(time_t val)
+time_t Creature::GetLastLeashExtensionTime() const
 {
-    if (val > 0)
-    {
-        if (_lastDamagedTime)
-            *_lastDamagedTime = val;
-        else
-            _lastDamagedTime = std::make_shared<time_t>(val);
-    }
-    else
-        _lastDamagedTime.reset();
+    return *GetLastLeashExtensionTimePtr();
 }
 
-void Creature::SetLastDamagedTimePtr(std::shared_ptr<time_t> const& val)
+void Creature::UpdateLeashExtensionTime()
 {
-    _lastDamagedTime = val;
+    (*GetLastLeashExtensionTimePtr()) = time(nullptr);
```

Comment:
Use `GameTime::GetGameTime().count()` instead of `time(nullptr)` (with time(nullptr) we are probably making a system call but GameTime::GetGameTime() has cached value - so no syscall).

Reply:
Added in https://github.com/azerothcore/azerothcore-wotlk/pull/19923.